libraries/Microsoft.Bot.Builder.Skills/Integration.AspNet.Core.Skills/SkillHandler.cs

```diff
@@ -351,33 +350,39 @@ protected override Task<ResourceResponse> OnSendConversationHistoryAsync(ClaimsI
         /// <returns>task with result.</returns>
         protected override Task<ResourceResponse> OnUploadAttachmentAsync(ClaimsIdentity claimsIdentity, string conversationId, AttachmentData attachmentUpload, CancellationToken cancellationToken = default)
         {
-            return InvokeChannelApiAsync<ResourceResponse>(_adapter, _bot, claimsIdentity, ChannelApiMethods.UploadAttachment, conversationId, attachmentUpload);
+            return HandleChannelRequestAsync<ResourceResponse>(claimsIdentity, ChannelApiMethods.UploadAttachment, conversationId, attachmentUpload);
         }
 
-        private async Task InvokeChannelApiAsync(BotAdapter adapter, IBot bot, ClaimsIdentity claimsIdentity, string method, string conversationId, params object[] args)
+        private static IInvokeActivity CreateSkillInvokeActivity(string callerServiceUrl, string callerConversationId)
```

Comment:
fine for now but we should avoid creating internal activities for wiring one part of framework to another